app/controllers/schedule_controller.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class ScheduleController < ApplicationController
+  def save
+    @document = Document.with_current_edition.find_by_param(params[:id])
+    @edition = @document.current_edition
+    current_revision = @edition.revision
+    date_issues = Requirements::DatetimeChecker.new(date_params).date_issues
+    datetime_issues = Requirements::DatetimeChecker.new(datetime_params).datetime_issues
+
+    if date_issues.any? || datetime_issues.any?
+      flash[:day_issues] = date_issues.items_for(:scheduled_day)
+      flash[:month_issues] = date_issues.items_for(:scheduled_month)
+      flash[:year_issues] = date_issues.items_for(:scheduled_year)
+      flash[:datetime_issues] = datetime_issues.items_for(:scheduled_datetime)
+
+      render "documents/show"
+      return
+    end
+
+    new_revision = current_revision.build_revision_update(
+      { scheduled_datetime: format_datetime },
+      current_user,
+    )
+    @edition.assign_revision(new_revision, current_user).save!
+
+    redirect_to document_path(@document)
+  end
+
+private
+
+  def format_datetime
+    year = params[:scheduled_year]
+    month = params[:scheduled_month]
+    day = params[:scheduled_day]
+    time = params[:scheduled_time]
+
+    Time.zone.parse("#{year}-#{month}-#{day}-#{time}")
+  end
+
+  def date_params
+    date_components = %I(day month year)
+    date_components.each_with_object({}) do |component, hash|
+      param = params["scheduled_#{component}".to_sym]
+      hash[component] = is_numeric?(param) ? param.to_i : param
```

Comment:
Could we let the checker handle this instead? That way, we can avoid having extra logic in the controller that's only used for the checker.

Reply:
Now moved into `ScheduledDatetimeChecker`